ccswbs/drupal-8-composer
# Add selenium to the pipeline

# To test
- Run pipeline on this branch in its default state (i.e. run.tests set to false)
   - _Expected result:_ Pipeline should run successfully with tests turned off.
- Run pipeline on this branch with run.tests set to true.
   - _Expected result:_ Selenium test for LDAP authentication should pass.
- Code review

# Changes
- Add selenium task to the pipeline (only triggered if run.tests pipeline variable is set to true)
    - Add generic appsettings.json file (with blank values that will be overridden during task)
    - Override appsettings.json file values for hostname, username, password in pipeline
    - Run tests on multidev created during deploy multidev step
- Only trigger builds for pull requests that use develop or master as a base branch (as opposed to all commits)
- Move drush into the test environment setup

(Also I'm going to squash all of the commits into a tidy commit once approved.)

# Current State vs Future State
- **Currently:** The test hostname, username, and password are currently environment variables from the build.
- **In future:** I would like us to grab the values from the Azure keyvault instead. I've included code for this which works, but is commented out currently because the keyvault firewall has not been configured to allow the Microsoft-hosted agent IP address through (note that the ip address ranges change weekly).
- I did try adding an ip address to the firewall as part of the pipeline, however, I was not able to find an ip address that the firewall was willing to accept (it was in a forbidden range apparently). If we can find the correct ip address on the fly, we might be able to use this option.
